tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py

```diff
@@ -0,0 +1,206 @@
+import pytest
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow_addons.callbacks import AverageModelCheckpoint
+from tensorflow_addons.optimizers import MovingAverage
+import os
+
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
+TRAIN_SAMPLES = 10
+NUM_CLASSES = 2
+INPUT_DIM = 3
+NUM_HIDDEN = 5
+BATCH_SIZE = 5
+EPOCHS = 5
+test_model_filepath = "test_model.h5"
+
+moving_avg = MovingAverage(
+    tf.keras.optimizers.SGD(lr=2.0), sequential_update=True, average_decay=0.5,
+)
+
+
+def get_data_and_model(optimizer=moving_avg):
+    x = tf.random.normal([TRAIN_SAMPLES, INPUT_DIM])
+    y = tf.random.normal([TRAIN_SAMPLES, NUM_CLASSES])
+    inputs = keras.layers.Input(INPUT_DIM)
+    layer1 = keras.layers.Dense(NUM_HIDDEN, input_dim=INPUT_DIM, activation="relu")(
```

Comment:
NIT: since you only have 1 layer maybe this variable to `hidden_layer` is better for readability.